transformer_engine/pytorch/module/layernorm_linear.py

```diff
@@ -638,6 +650,9 @@ class LayerNormLinear(TransformerEngineBaseModule):
                              together with the output of the linear transformation.
                              Example use case: residual connection for transformer module is
                              taken post layernorm.
+    return_layernorm_output_gathered : bool, default = `False`
+                             if set to `True`, output of layernorm is returned after the all
+                             gather operation. Ignored if return_layernorm_output is False
```

Comment:
Let's also add the support for this in `LayerNormMLP` + explain the use case more thoroughly in the docs and mention SP

Reply:
okay do!